lib/fs.js

```diff
@@ -452,7 +452,7 @@ fs.readFile = function(path, options, callback) {
   callback = maybeCallback(callback || options);
   options = getOptions(options, { flag: 'r' });
 
-  if (handleError((path = getPathFromURL(path)), callback))
+  if (handleError((path = getPathFromURL(path, true)), callback))
```

Comment:
Can you update the `validatePath` function to check the trailing slash for strings?